sql/parse/parse.go

```diff
@@ -83,6 +84,32 @@ func Parse(ctx *sql.Context, query string) (sql.Node, error) {
 	return convert(ctx, stmt, s)
 }
 
+//TODO: Move this to somewhere else perhaps?
+func parseShowCreateTable(s string) (sql.Node, error) {
+	r := bufio.NewReader(strings.NewReader(s))
+
+	var table string
+	steps := []parseFunc{
+		expect("show create table"),
+		skipSpaces,
+		readIdent(&table),
+		skipSpaces,
+		checkEOF,
+	}
+
+	for _, step := range steps {
+		if err := step(r); err != nil {
+			return nil, err
+		}
+	}
+
+	return plan.NewShowCreateTable(
+		&sql.UnresolvedDatabase{},
```

Comment:
You can change the `assign_catalog` rule to also give the current database to `ShowCreateTable` nodes. You have an example here: https://github.com/src-d/go-mysql-server/blob/master/sql/analyzer/assign_catalog.go#L22

But, anyway, I think your approach is actually better than that since you're gonna need access to the database itself.

IMHO, it's ok as it is now.

Reply:
I just pushed the changes